dna/problems.py

```diff
@@ -321,10 +321,24 @@ def plot(self, predictions, targets, scores, plot_dir: str):
             predicted_ranks = np.array(predicted_ranks)
             actuals = merged_data['test_f1_macro'].tolist()
             actual_ranks = utils.rank(actuals)
+
+            # Average all the scores at k to make the plot title a reasonable length
             group_scores = scores_by_group[dataset_id]
+            group_scores = self.shorten_k_rank_scores(group_scores)
+
             plot_name = dataset_id + '_plot'
             super()._plot_base(predicted_ranks, actual_ranks, plot_name, plot_dir, group_scores, type(self).__name__)
 
+    @staticmethod
+    def shorten_k_rank_scores(rank_scores: dict):
+        """Takes the average of the top k, k regret, and ndcg rank scores so the score at every k isn't reported"""
+
+        rank_scores = rank_scores.copy()
+        rank_scores['ndcg_at_k'] = np.mean(rank_scores['ndcg_at_k'])
+        rank_scores['regret_at_k'] = np.mean(rank_scores['regret_at_k'])
+        rank_scores['n_correct_at_k'] = np.mean(rank_scores['n_correct_at_k'])
```

Comment:
this is not needed. the total scores of the at_k metrics already have this information.

Reply:
The total scores are not available at this part. We are plotting by dataset. Only the information of the dataset at a given iteration in the for loop is available. We need to do this to get the average of the at k scores for this particular data set. If we deleted these lists, that info would not show up on the plots.

Reply:
`scores` is available inside the loop. You can access the total scores from there.

Reply:
That's the average at k scores across all data sets. The plot should contain the at k scores for its corresponding dataset

Reply:
Ah, I see my mistake now. You are right.